### PR TITLE
Add ios date input picker

### PR DESCRIPTION
**Pull request by KarinaDavtyan**
Also refactored PassengerForm so the componentDidUpdated is not needed anymore as DateInput.js would lift state once date composed (not on button click as before).

### CODE REVIEW

File: apps/core/components/passengerForm/PassengerForm.js [hunk 1]
```diff
@@ -94,24 +92,20 @@ class PassengerForm extends React.Component<Props, State> {
     // If is in edit mode preload passenger data into the form state
     if (editModeOpened) {
       if (props.prefillData) {
+        const dateOfBirth =
```
Comment (aalises) [inline apps/core/components/passengerForm/PassengerForm.js]: maybe you can destructure `const { prefillData } = props.prefillData`, as it's accessed quite a lot

File: packages/components/src/dateInput/DateInput.js [hunk 2]
```diff
@@ -67,13 +65,15 @@ export default class DateInput extends React.Component<Props, State> {
...
 
-  componentDidUpdate(prevProps: Props) {
-    const { isDateRequested, onDateChange } = this.props;
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    const { onDateChange } = this.props;
     const { day, month, year } = this.state;
-    if (isDateRequested !== prevProps.isDateRequested && isDateRequested) {
-      if (day !== '' && year !== '') {
+    if (day !== prevState.day || year !== prevState.year) {
+      if (day.length === 2 && year.length === 4) {
```
Comment (aalises) [inline packages/components/src/dateInput/DateInput.js]: We had constants for that? `MAX_DAY_LENGTH`, etc...

Comment (KarinaDavtyan) [reply packages/components/src/dateInput/DateInput.js]: Yes we have, forgot about them :(

Comment (KarinaDavtyan) [reply packages/components/src/dateInput/DateInput.js]: fixed

File: apps/core/components/passengerForm/PassengerForm.js (hunk 1 above)
Comment (KarinaDavtyan) [reply apps/core/components/passengerForm/PassengerForm.js]: fixed, thanks!

**Pull request merged**